library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -742,7 +742,7 @@ private void updateDatabaseFromHeaders(State state) {
         if (state.mimeType != null) {
             values.put(DownloadContract.Downloads.COLUMN_MIME_TYPE, state.mimeType);
         }
-        values.put(DownloadContract.Downloads.COLUMN_TOTAL_BYTES, originalDownloadInfo.getTotalBytes());
+        values.put(DownloadContract.Downloads.COLUMN_TOTAL_BYTES, state.totalBytes);
```

Comment:
the method ordering makes this look strange, but at the is point the `state.totalSize` has already been set